app/code/Magento/Checkout/view/frontend/web/js/sidebar.js

```diff
@@ -310,6 +310,10 @@ define([
 
                     if (response.success) {
                         callback.call(this, elem, response);
+                        var currentURL = window.location.pathname;
+                        if (currentURL.includes("/checkout/cart/")) {
+                            location.reload();
```

Comment:
Nope, such behavior is unacceptable. Please read comments in corresponding issue more carefully.

Needed page content must be re-rendered via AJAX.

Reply:
Hello @orlangur 
As per Expected result issue, we need to reload the page because only mini cart load data using ajax,
but for update same change in  checkout/cart/index page, we need reload page

Reply:
@ravi-chandra3197 no.

Reply:
@ravi-chandra3197 expected result is now fixed to avoid confusion

Reply:
Hello @orlangur 
For update mini cart change in cart page, we need to reload checkout/cart/index because it's not a problem to just update product but also subtotal, Discount Code and additional info like shipping charges, so we need to reload cart page instead of   re-rendered cart page small part

Reply:
@ravi-chandra3197 full page reload is not acceptable. Reload only what is needed to be reloaded.